systems/plants/RigidBodyCylinder.m

```diff
@@ -103,6 +103,19 @@ function writeWRLShape(obj,fp,td)
       
       capsule = RigidBodyCapsule(obj.radius,obj.len,obj.T);
     end
+    
+    function h = draw(obj,model,kinsol,body_ind)
+      persistent cylinder_pts; % for all cylinders
+      if isempty(cylinder_pts)
+        [x,y,z] = cylinder;
+        cylinder_pts = [x(:)'; y(:)'; z(:)'-.5; 1+0*x(:)'];
+      end
+      
+      pts = forwardKin(model,kinsol,body_ind,obj.T(1:3,:)*diag([obj.radius,obj.radius,obj.len,1])*cylinder_pts);
```

Comment:
it's not actually repeated, because it has the scaling by radius/length embedded.

On Apr 15, 2015, at 8:08 AM, Andres Valenzuela <notifications@github.com> wrote:

> In systems/plants/RigidBodyCylinder.m:
> 
> > @@ -103,6 +103,19 @@ function writeWRLShape(obj,fp,td)
> >        
> >        capsule = RigidBodyCapsule(obj.radius,obj.len,obj.T);
> >      end
> > +    
> > +    function h = draw(obj,model,kinsol,body_ind)
> > +      persistent cylinder_pts; % for all cylinders
> > +      if isempty(cylinder_pts)
> > +        [x,y,z] = cylinder;
> > +        cylinder_pts = [x(:)'; y(:)'; z(:)'-.5; 1+0*x(:)'];
> > +      end
> > +      
> > +      pts = forwardKin(model,kinsol,body_ind,obj.T(1:3,:)*diag([obj.radius,obj.radius,obj.len,1])*cylinder_pts);
> Should this section of repeated code move up into the parent class?
> 
> —
> Reply to this email directly or view it on GitHub.
>